tsfile/src/main/java/org/apache/iotdb/tsfile/utils/PublicBAOS.java

```diff
@@ -76,12 +76,12 @@ public void writeTo(OutputStream out) throws IOException {
    * reusing the already allocated buffer space.
    */
   @Override
-  public void reset() {
+  public synchronized void reset() {
```

Comment:
the same, we remove the synchronized in all override method in PublicBAOS

Reply:
Got it, I prefer to post this link here in this function, so that other developers know about the problem, otherwise, sonar will report a bug, and other developers may fix it.